### PR TITLE
pushsync: split sync to chunksWorker and receiptsWorker

### PR DESCRIPTION
**Pull request by janos**
These are cherry picked changes on pusher from obsolete PR https://github.com/ethersphere/swarm/pull/1857.

### CODE REVIEW

File: pushsync/pusher_test.go [hunk 1]
```diff
@@ -160,9 +160,9 @@ func (lb *loopBack) Register(topic string, _ bool, handler func(msg []byte, p *p
 // Send publishes a msg with a topic and directly calls registered handlers with
 // that topic
 func (lb *loopBack) Send(to []byte, topic string, msg []byte) error {
-	if !delayResponse() {
-		return nil
-	}
+	// if !delayResponse() {
```
Comment (acud) [inline pushsync/pusher_test.go]: delete this?

File: pushsync/pusher.go [hunk 2]
```diff
@@ -146,49 +153,11 @@ func (p *Pusher) sync() {
...
-			tag := item.tag
-			if tag != nil {
-				tag.Inc(chunk.StateSynced)
-				if tag.Done(chunk.StateSynced) {
-					p.logger.Debug("closing root span for tag", "taguid", tag.Uid, "tagname", tag.Name)
-					tag.FinishRootSpan()
-				}
-				// finish span for pushsync roundtrip, only have this span if we have a tag
-				item.span.Finish()
+			//go func(ch chunk.Chunk) {
```
Comment (acud) [inline pushsync/pusher.go]: commented code

File: pushsync/pusher.go [hunk 3]
```diff
@@ -72,27 +77,36 @@ type pushedItem struct {
...
+	case <-p.closedReceipts:
+		select {
+		case <-p.closedChunks:
+			// all closed
+			return
+		case <-timer:
+		}
+	case <-timer:
 	}
+	log.Error("timeout closing pusher")
```
Comment (acud) [inline pushsync/pusher.go:109]: ?

Comment (janos) [reply pushsync/pusher.go:109]: The closing timer is reached at this point.

**Review by nonsense — APPROVED** [review]

**Comment by nonsense** [comment]
Great stuff, thanks, Janos.

Once this gets merged, we should benchmark `master`, and see how to amend the `sendChunkMsg` call, as PSS is still non-blocking, and we are not handling the error returned.

**Comment by janos** [comment]
Thank you, Anton. This is a joint effort. :)

**Review by nonsense — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**